src/include/ndpi_main.h

```diff
@@ -25,6 +25,10 @@
 #define __NDPI_MAIN_H__
 
 #include "ndpi_includes.h"
+#ifdef NDPI_LIB_COMPILATION
+/* for macros NDPI_LOG_* in ndpi_define.h */
+#include "ndpi_config.h"
+#endif
```

Comment:
@lucaderi, could you take a look, please? You removed a similar include in https://github.com/ntop/nDPI/commit/246d115d7bf126c284df739c35cd9b790bf140cf to "prevents ndpi_config.h to be included everywhere in apps using nDPI that might leade to #define redefinitions".
Could you check if this fix is fine or if it triggers some regressions?

Reply:
If we build the library with the "--enable-debug-messages" option, then we want debug printing to work.
Perhaps this can be done using configure.ac by adding -DNDPI_ENABLE_DEBUG_MESSAGES to CFLAGS.
There is another hidden problem: the size of the ndpi_detection_module_struct structure depends on this option.
Let's leave the "user_data" field unconditional. 8 bytes in 250 kilobytes won't change anything.

Reply:
You patch clearly fix the debug log. My only concern is if it breaks whatever @lucaderi  fixed in https://github.com/ntop/nDPI/commit/246d115d7bf126c284df739c35cd9b790bf140cf.

> There is another hidden problem: the size of the ndpi_detection_module_struct structure depends on this option.
> Let's leave the "user_data" field unconditional. 8 bytes in 250 kilobytes won't change anything.

No strong opinion on that

Reply:
For me we can accept it, try to compile on all platforms and see what happens.